spring-vault-core/src/main/java/org/springframework/vault/authentication/MacAddressUserId.java

```diff
@@ -128,7 +129,7 @@ public String createUserId() {
 							"Cannot determine NetworkInterface"));
 		}
 		catch (IOException e) {
-			throw new IllegalStateException(e);
+			throw new VaultClientException("While creating userId", e);
```

Comment:
Rather `VaultException`.

Reply:
It's definitely an issue unquestionably on the client ;)

Reply:
The client is `RestTemplate` and `VaultTemplate`.